app/utils/text_processing.py

```
@@ -131,94 +137,419 @@ def remove_stopwords(tokens: list, language:str ='en'):
+            return await smart_keyword_extraction(text, user_id, max_keywords, db_session)
+    
+    # 1. 기본 키워드 추출
+    basic_keywords = extract_keywords(text, max_keywords * 2)
+    
+    if not basic_keywords:
         return []
+    
+    # 2. 사용자의 기존 키워드 정보 가져오기 (PostgreSQL에서)
+    try:
+        # 사용자의 문서 통계 가져오기
+        user_stats = await vector_service.get_user_document_stats(session, user_id)
```

Ruff (0.11.9)

Local variable `user_stats` is assigned to but never used

Remove assignment to unused variable `user_stats`

(F841)